tests/test_models_layers.py

```diff
@@ -187,3 +187,34 @@ def forward(self, x: LayerRef) -> LayerRef:
 
   assert "TestBlock" in net_dict and "TestBlock_0" in net_dict
   assert_equal(net_dict["TestBlock_0"]["reuse_params"], "TestBlock")
+
+
+def test_multiple_returns():
+  class _SubNet(Module):
+    def __init__(self):
+      super().__init__()
+      self.linear = Linear(n_out=42, activation=None)
+
+    def forward(self, x: LayerRef) -> Tuple[LayerRef, LayerRef]:
+      """
+      Forward
+      """
+      x = self.linear(x)
+      return x, x
+
+  class _Net(Module):
+    def __init__(self):
+      super().__init__()
+      self.sub = _SubNet()
+
+    def forward(self) -> LayerRef:
+      """
+      Forward
+      """
+      x = get_extern_data("data")
+      out, add_out = self.sub(x)
+      assert out == add_out[0]
+      return out
+
+  net = _Net()
+  net.make_root_net_dict()
```

Comment:
Also store the dict, and print it, verify it by hand (by looking at it), and then add some sanity checks.
```suggestion
  net_dict = net.make_root_net_dict()
  pprint(net_dict)
```